### PR TITLE
Cq 4348947

### PR DESCRIPTION
**Pull request by barshat7**
## Description

If Screen Reader Preference is set by the author, then corresponding aria-label will be set for the component.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java [hunk 1]
```diff
@@ -46,6 +46,10 @@ public abstract class AbstractOptionsFieldImpl extends AbstractFieldImpl impleme
     @Nullable
     private String[] enumNames;
 
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL, name = "custom")
```
Comment (nit23uec) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:49]: why name is set as custom in annotations?

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java [hunk 2]
```diff
@@ -177,16 +177,20 @@ public String getScreenReaderText() {
     @Nullable
     public String getHtmlScreenReaderText() {
         // this can be used in sightly to compute initial html
-        String screenReaderText = getName();
+        String screenReaderText = "";
         if (AssistPriority.LABEL.equals(assistPriority)) {
```
Comment (nit23uec) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java:181]: is there a default assist priority value desired now? It does not seems so from this PR.
we should have a default value for screen reader text.

**Comment by nit23uec** [comment]
why do we need `aria-label` in all core components?
it's required only when there is no appropriate text visible in the DOM that could be referenced as a label. Do we have any such core component?

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java (hunk 2 above)
Comment (barshat7) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java:181]: Default is empty. (No corresponding property is saved in JCR)

**Comment by barshat7** [comment]
> why do we need `aria-label` in all core components? it's required only when there is no appropriate text visible in the DOM that could be referenced as a label. Do we have any such core component?

Screenreader preference could be different than the <label> e.g Description or Custom Text, as we won't be able to change the label of the field, we are resorting to ARIA attributes. I think this is on par with V1. @rismehta

**Pull request closed without merging**